### PR TITLE
fix(apigatewayv2): error while configuring ANY as an allowed method in CORS

### LINKED ISSUE #13280: (apigatewayv2): support '*' as a valid HTTP method
Hi @nija-at , I don't think this is just a limitation of API Gateway but rather a mismatch between what is expected by API Gateway and what is exposed by CDK.

API gateway accepts `*` to represent any HTTP method. The HttpMethod enum doesn't have a value for that.
The following code synthesises and deploys correctly:
```
allowMethods: ["*"] as any
```

I would suggest to either create a separate enum for HTTP methods that aligns with what API gateway expects, or to internally map "ANY" to "*" during synthesis. It's wrong to claim that `allowMethods` accepts an array of `HttpMethod` when in fact not all possible values would be correct.

_Originally posted by @lorenzo2897 in https://github.com/aws/aws-cdk/issues/10230#issuecomment-767720313_

-----

Lack of this method makes it difficult to configure CORS preflight. See https://github.com/aws/aws-cdk/issues/10230

### LINKED ISSUE #13643: (aws-apigatewayv2): Specifying HttpMethod.ANY as allowMethod of CORS configuration causes deployment failure.
```typescript
new apigatewayv2.HttpApi(this, 'API', {
  corsPreflight: {
    allowMethods: [apigatewayv2.HttpMethod.ANY],
    allowOrigins: ['*'],
    allowHeaders: ['*'],
    maxAge: cdk.Duration.seconds(600),
  },
});
```

This generates such template:

```yaml
"APIXXXXXX": {
  "Type": "AWS::ApiGatewayV2::Api",
  "Properties": {
    "CorsConfiguration": {
      "AllowMethods": [
        "ANY"
      ],
      ...
```

But deploying this fails with the error: `Method ANY is not supported (Service: AmazonApiGatewayV2; Status Code: 400; Error Code: BadRequestException; ...)`

### What actually happened?

The below is correct CloudFormation template:

```yaml
"APIXXXXXX": {
  "Type": "AWS::ApiGatewayV2::Api",
  "Properties": {
    "CorsConfiguration": {
      "AllowMethods": [
        "*"
      ],
      ...
```

So the beginning code should generate such YAML.

### Environment

  - **CDK CLI Version:2.0.0-alpha.0 (build beadc66)**
  - **Framework Version: 2.0.0-alpha.0**
  - **Node.js Version: v15.10.0** 
  - **OS               : Linux acm-envy2-win-wsl 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64 GNU/Linux**
  - **Language (Version): TypeScript 4.2.3** 

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by robertd**
API gateway expects `*` to represent `ANY` HTTP method. `HttpMethod` enum doesn't 
have a value for that, thus causing errors when passing  `HttpMethod` 's `ANY` option.

This commit introduces `CorsHttpMethod` enum which has a proper `ANY` mapping to `*`.

Closes #13280.
Closes #13643.

BREAKING CHANGE: The type of `allowMethods` property under `corsPreflight`
section is changed from `HttpMethod` to `CorsHttpMethod`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class HttpRouteKey {
     if (path !== '/' && (!path.startsWith('/') || path.endsWith('/'))) {
       throw new Error('path must always start with a "/" and not end with a "/"');
     }
-    return new HttpRouteKey(`${method ?? 'ANY'} ${path}`, path);
+    return new HttpRouteKey(`${method ?? HttpMethod.ANY} ${path}`, path);
```
Comment (robertd) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts:64]: @nija-at This was hardcoded to `ANY` so I've changed and I've decided to change it to use enum instead. Kinda our fo the scope for this PR but somewhat related. Let me know if that's ok.

**Comment by robertd** [comment]
@nija-at Please review when you get a moment. Thanks!

**Comment by nija-at** [comment]
Hello - 

Sorry about the delay in reviewing PRs. We are experiencing an increased backlog of items that need our attention.
We will get to your PR as soon as we are able to. Apologies for the delay.

**Comment by robertd** [comment]
> Hello -
> 
> Sorry about the delay in reviewing PRs. We are experiencing an increased backlog of items that need our attention.
> We will get to your PR as soon as we are able to. Apologies for the delay.

No worries. Thanks for letting me know.

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts (hunk 1 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts:64]: Yep, that's ok 👍

**Review by nija-at — CHANGES_REQUESTED** [review]
Code changes look good to me.

I've fixed up the title and the 'BREAKING CHANGE' clause to more closely align with how we usually announce it.

Can you update the commit description (above the breaking changes clause and 'closes' statements)? This will go into the description of the 'git commit', so use the appropriate language. 

Usually, the description should answer the two questions - 'Why is this change necessary?', 'How does this commit address the issue?'

**Comment by robertd** [comment]
@nija-at Done. Let me know if there is anything else I need to do. Thanks!

**Comment by robertd** [comment]
I've updated and force-pushed git commit title too.

**Review by nija-at — APPROVED** [review]

**Pull request merged**